src/internal/utils/date-time/format-date-localized.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export default function formatDateLocalized({
+  date: isoDate,
+  hideTimeOffset,
+  isDateOnly,
+  timeOffset,
+  locale,
+}: {
+  date: string;
+  hideTimeOffset?: boolean;
+  isDateOnly: boolean;
+  timeOffset?: number;
+  locale?: string;
+}) {
+  const date = new Date(isoDate);
+
+  const formattedDate = new Intl.DateTimeFormat(locale, {
+    day: 'numeric',
+    month: 'long',
+    year: 'numeric',
+  }).format(date);
+
+  if (isDateOnly) {
+    return formattedDate;
+  }
+
+  const formattedTime = new Intl.DateTimeFormat(locale, {
+    hour: '2-digit',
+    hourCycle: 'h23',
+    minute: '2-digit',
+    second: '2-digit',
+  }).format(date);
+
+  const formattedDateTime = formattedDate + getDateTimeSeparator(locale) + formattedTime;
+
+  if (hideTimeOffset) {
+    return formattedDateTime;
+  }
+
+  const formattedTimeOffset = formatTimeOffset(isoDate, timeOffset);
+  return formattedDateTime + ' ' + formattedTimeOffset;
+}
+
+function formatTimeOffset(isoDate: string, offsetInMinutes?: number) {
```

Comment:
we have similar logic here: https://github.com/cloudscape-design/components/blob/main/src/internal/utils/date-time/format-timezone-offset.ts

Can we try avoiding duplications?

Reply:
Note that there are two functions because one localizes the time offset for the ISO format (e.g, `+01:00`) and the other one for the human-readable format (e.g, `(UTC+1)`).

But I now refactored this to make things hopefully a bit more clear. The old `formatTimezoneOffset` function is now `formatTimeOffsetISO`, and the new one is `formatTimeOffsetLocalized`. They are in the same file and the common bits are extracted into small functions.

Also renamed the file `format-timezone-offset.ts` to `format-time-offset.ts`. Time zone (as opposed to time offset) does not play a role in these functions, so I think it's better to avoid mixing both concepts unnecessarily.